index.bs

```diff
@@ -3199,6 +3203,129 @@ The [=remote end steps=] with |command parameters| are:
 1. Return a new map matching the <code>ScriptEvaluateResultSuccess</code>
    production, with the <code>result</code> field set to |result|.
 
+#### The script.callFunction Command ####  {#command-script-callFunction}
+
+The <dfn export for=commands>script.callFunction</dfn> command calls a provided
+function with given arguments in a given realm.
+
+ <code>RealmInfo</code> can be either a realm or a browsing context.
+
+Issue: declare `RemoteValueArgument`.
+
+Issue: declare `LocalValueArgument`.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      ScriptCallFunctionCommand = {
+        method: "script.callFunction",
+        params: ScriptCallFunctionParameters
+      }
+
+      ScriptCallFunctionParameters = {
+        functionDeclaration: text;
+        arguments: [CallFunctionArgument];
+        ?awaitPromise: bool;
+        target: Target;
+      }
+
+      CallFunctionArgument = (
+        RemoteValueArgument //
+        LocalValueArgument
+      );
+
+      RemoteValueArgument = {
+        objectId: string;
+      };
+
+      LocalValueArgument = {
+        type: string;
+        value: any;
+      };
+
+      </pre>
+   </dd>
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      ScriptCallFunctionResult = (
+        ScriptCallFunctionResultSuccess //
+        ScriptCallFunctionResultException
+      )
+
+      ScriptCallFunctionResultSuccess = {
+         result: RemoteValue
+      }
+
+      ScriptCallFunctionResultException = {
+        exceptionDetails: ExceptionDetails
+      }
+    </pre>
+   </dd>
+</dl>
+
+Issue: TODO: Add timeout argument as described in the script.evaluate.
+
+The [=remote end steps=] with |command parameters| are:
+
+1. Let |realm| be the result of [=trying=] to [=get a realm from a target=]
+   given the value of the <code>target</code> field of |command parameters|.
+
+1. Let |environment settings| be the [=environment settings object=] whose
+   [=realm execution context=]'s Realm component is |realm|.
+
+1. [=Prepare to run script=] with |environment settings|.
+
+1. Let |function declaration| be the value of the
+   <code>functionDeclaration</code> field of |command parameters|.
+
+1. Let |arguments declaration| be the value of the <code>arguments</code> field
+   of |command parameters|.
+
+1. Let |JS arguments list| be an empty list.
+
+1. For each |arg declaration| of |arguments declaration|:
+
+  Issue: describe |deserialize CallFunctionArgument|.
+
+  1. Let |JS arg| be the result of
+     |deserialize CallFunctionArgument|(|arg declaration|).
+
+  1. Append |JS arg| to the |JS arguments list|.
+
+1. Let |JS function object| be the result of
+   [=InstantiateOrdinaryFunctionObject=](|function declaration|).
```

Comment:
So if I understand ECMAScript correctly, this only works for the `function [name]() {}` form and not, for example `() => {}`. I also wonder if we should support async functions here.

`InstantiateOrdinaryFunctionObject` also requires explicitly setting the scope of the function, which seems to be missing here.

Reply:
Valid point. 
It looks like it would make more sense just to evaluate the `functionDeclaration`, [assert the result is `callable`](https://tc39.es/ecma262/#sec-iscallable), and [ [[call]] ](https://tc39.es/ecma262/#sec-ecmascript-function-objects-call-thisargument-argumentslist) with deserialized arguments.